test/plugin/base_test.rb

```diff
@@ -612,6 +612,30 @@ def test_structured_payload_json_log_detect_json_container_parsed
       end
   end
 
+  # Verify that when the log has only one effective field (named 'log',
+  # 'message' or 'msg') and the field is in JSON format, the field is parsed as
+  # JSON and sent as jsonPayload.
```

Comment:
What should happen if more than one of msg, log, or message are defined in the input? Is that possible?

Reply:
It is possible. If that is the case, we will not send the field (named 'msg', 'log' or 'message') as `jsonPayload`. Instead, it will be treated as a `textPayload`. There is some inconsistent behavior regarding to `detect_json`. We have a separate ticket for it: b/119681841.

Reply:
Do we have a test for that behavior? If not, should we add one?

Reply:
If more than one field is present in the original record, wouldn't JSON detection not kick in at all?